### PR TITLE
Add README.md with rpm-build instructions #2

### LINKED ISSUE #2: Add README.md with rpm build instructions using official installer
Our official downloadable installer, itself built using the https://github.com/rockstor/rockstor-installer repo, has all rpm build prerequisites preinstalled; bar currently wget, to be addressed in https://github.com/rockstor/rockstor-installer/issues/133 "Proposal to add wget to our included packages".

It is proposed that we base our README.md contents around in-place rpm rebuilds to enable complete end-to-end testing if code changes in any relevant issue also require modifications in this repo. We also then have an easier path for domain experts to contribute improvements in this our initial Poetry build approach.

We should also mention in this repo-heading documentation that we mirror the master/testing branch structure of rockstor-core for, respectively the Stable and Testing channel updates. We are however on RC4 currently so will initially only have a testing branch in this repo with any significant contents.

### PR DESCRIPTION
**Pull request by phillxnet**
Initial draft of developer focused About/Howto README.md. Intended to enable contributors to do DIY RPM builds for end-to-end testing of proposed changes in related repositories. Based on using the official Rockstor installer to promote self-hosting capabilities.

Fixes #2

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+
+RPM Relocation; see: "Prefix: /opt" in the spec file, is known to be incompletely implemented:
+both here in [rockstor-rpmbuild](https://github.com/rockstor/rockstor-rpmbuild)
+and in the [rockstor-core](https://github.com/rockstor/rockstor-core).
+Contributions to complete this capability are of course welcome.
+But this is not a priority for the project.
+
+### Package renaming
+
+There is facility within the rpmbuild spec file to rename the resulting package.
```
Comment (Hooverdan96) [inline README.md:128]:
```suggestion
There is a facility within the rpmbuild spec file to rename the resulting package.
```

**Review by Hooverdan96 — COMMENTED** [review]
Just a few cosmetic suggestions (sometimes I am not sure whether it's a "british" English or "American" English thing or my lack of mastery of the English language :) , so take these with a grain of salt.

File: README.md [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+zypper install --no-recommends rpm-build
+```
+**NOTE: Solution 1: deinstallation of busybox-gzip ...**
+May be required (replaced by the full gzip).
+See: [pre-install gzip to avoid busybox-gzip via dependency](https://github.com/rockstor/rockstor-installer/issues/135)
+
+
+## Build the RPM/SRPM
+
+Both the regular RPM, and it's source variant (SRPM) can be build with a single command.
```
[the remark this answers is not in the document]
Comment (phillxnet) [reply README.md:83]: Oh no, not again
@Hooverdan96 thanks. I can't believe I'm still doing this. Anyway, I'll commit this quietly before @FroggyFlox see yet another of this exact same mistake from me. Maybe there's a 'thing' I can run (like my brain) before actually pushing anything to check just for this seemingly perpetual flaw in my its/it's user.

**Comment by phillxnet** [comment]
@Hooverdan96 Thanks for the excellent review. As always much appreciated.

All suggestions gladly accepted, I'll go ahead an merge this so we have it for more general use.

Do feel free to pop in and make any more modifications enhancements by the way. I was thinking initially of extending the 'getting the spec file' section with advise on building from the SRPMs we publish in our repos. But that is less orientated towards contribution and there were enough mistakes in what I did initially: so yes - for another day (pr): maybe. Plus it involves other commands/locations/caveats and keeping this file fairly small is key also I think.

**Pull request merged**